test/back_end/api_post_test.spec.js

```diff
@@ -110,6 +110,49 @@ describe("Test POST/UPDATE/DELETE requests to API", function() {
             }
         });
 
+        it("launch the survey with survey_id = 1", async function () {
+            try {
+                let fetched = await fetch(
+                    "http://localhost:3000/post_info.php?what=launch_survey&user_id=woods13&survey_id=1&action=add_or_update"
+                );
+                expect(fetched).to.have.status(200);
+                let fetchedJSON = await fetched.json();
+                expect(fetchedJSON.TYPE).to.be.equal("survey_package");
+                expect(parseInt(fetchedJSON.DATA.survey_instance_id)).to.satisfy(Number.isInteger);
+                let survey_instance_id = fetchedJSON.DATA.survey_instance_id;
+                let override_token = fetchedJSON.DATA.override_token;
+                fetched = await fetch(
+                    "http://localhost:3000/get_info.php?what=survey_results&user_id=woods13&course=UofT&target_ta=woods13&term=201709"
+                );
+                fetchedJSON = await fetched.json();
+                // Test whether the returned survey instance contains the survey_instance we just add
+                let data_package = fetchedJSON.DATA;
+                let contain_survey_instance_id = false;
+                for (let obj of data_package) {
+                    if (obj.survey_instance_id == survey_instance_id) {
+                        contain_survey_instance_id = true;
+                    }
+                }
+                expect(contain_survey_instance_id).to.be.true;
+            } catch (error) {
+                throw error;
+            }
+        });
+
+        it("launch the survey with survey_id = 5", async function () {
+            try {
+                let fetched = await fetch(
+                    "http://localhost:3000/post_info.php?what=launch_survey&user_id=woods13&survey_id=5&action=add_or_update"
+                );
+                expect(fetched).to.have.status(200);
+                let fetchedJSON = await fetched.json();
+                expect(fetchedJSON.TYPE).to.be.equal("survey_package");
+                expect(parseInt(fetchedJSON.DATA.survey_instance_id)).to.satisfy(Number.isInteger);
```

Comment:
How about you launch it twice and make sure the instance numbers are different but the questions are the same. It also would be a good idea to make sure that the questions in the launched version are the same as the questions in the unlaunched one.

Reply:
For now, because of the `get` API is not ready yet, I cannot keep track of the questions in the `survey_instances`. But I can do the launching twice and make sure the instances_id are different.